skglm/utils.py

```diff
@@ -80,6 +80,49 @@ def prox_2_3(x, u):
     return res
 
 
+@njit
+def sigmoid(x):
+    """Vectorwise sigmoid."""
+    out = 1 / (1 + np.exp(-x))
+    return out
+
+
+@njit
+def weighted_dot(X, b, weights, j, ignore_b=False):
+    """Weighted dot product between X[:, j] and b."""
+    res = 0.
+    if ignore_b:
+        # if ignore_b, weighted dot product between X[:, j] and X[:, j]
+        for i in range(X.shape[0]):
+            res += (X[i, j] ** 2) * weights[i]
+    else:
+        for i in range(X.shape[0]):
```

Comment:
you can dot the dot * and .sum no ? and remove this function. I used it in Cython